### PR TITLE
Integration tests

### PR DESCRIPTION
**Pull request by GinaStavropoulou**
This mainly has 3 integration tests, for common, party_walls and floors_estimation.

What else:

- Removal of all BGT usage
- Some formatting (sorry)
- Corrected a few things to remove deprecation warnings
(see comments below)

The most relevant work is in the test_integration.py files. 
I basically mocked upstream assets and run the rest of the jobs - not sure if there is any better way to mock upstream assets so feel free to change as you see fit. 
Specifically for the pipeline jobs, the integration runs only the nl_reconstruction and nl_export jobs. I did not manage to add nl_deploy yet but I might write a separate test for it later. 

You can clone the repo and follow the instructions in README to run all the tests but you will need tyler, geoflow etc. - so maybe do that in gilfoyle. 

Before merging, pyproject.toml files need to be updated again to remove references to this branch.

### CODE REVIEW

File: README.md
Comment (GinaStavropoulou) [inline README.md:13]: Please see here if we are missing anything

File: docker/postgres/Dockerfile
Comment (GinaStavropoulou) [inline docker/postgres/Dockerfile:3]: MAINTAINER was deprecated. I also added Balázs here since the previous docker file was very similar.

File: packages/common/src/bag3d/common/resources/__init__.py
Comment (GinaStavropoulou) [inline packages/common/src/bag3d/common/resources/__init__.py:29]: This was not working for me

File: packages/common/src/bag3d/common/utils/requests.py
Comment (GinaStavropoulou) [inline packages/common/src/bag3d/common/utils/requests.py:53]: `i had to add this for avoiding ssl verification

File: makefile
Comment (GinaStavropoulou) [inline makefile:50]: added this to slow, fast and integration tests (only on gilfoyle)

File: packages/common/tests/test_database.py
Comment (GinaStavropoulou) [inline packages/common/tests/test_database.py:83]: this test is still not working so I remove it from the coverage.

File: packages/core/pyproject.toml
Comment (GinaStavropoulou) [inline packages/core/pyproject.toml:23]: should be changed before merging.

File: packages/core/src/bag3d/core/assets/ahn/core.py
Comment (GinaStavropoulou) [inline packages/core/src/bag3d/core/assets/ahn/core.py:145]: this function seems to be unused in the whole repo - so I removed it from the coverage. Should it be removed?

File: packages/core/src/bag3d/core/assets/export/archive.py
Comment (GinaStavropoulou) [inline packages/core/src/bag3d/core/assets/export/archive.py:96]: I had to remove existing zip files, because the tests were not running for a second time (the were not overwriting the file)

File: packages/core/src/bag3d/core/assets/export/metadata.py
Comment (GinaStavropoulou) [inline packages/core/src/bag3d/core/assets/export/metadata.py:186]: I had an issue here where the 'event_record_list' was None for the test but I wanted it to keep going even with this being None. However, we probably need to add some warning or some error handling. What do you think?

File: packages/core/src/bag3d/core/jobs.py
Comment (GinaStavropoulou) [inline packages/core/src/bag3d/core/jobs.py:15]: Deprecation warning

File: packages/core/tests/test_assets_bag.py
Comment (GinaStavropoulou) [inline packages/core/tests/test_assets_bag.py:15]: just formatting

File: packages/core/tests/test_assets_input.py
Comment (GinaStavropoulou) [inline packages/core/tests/test_assets_input.py:36]: Added this test

File: packages/core/tests/test_integration.py
Comment (GinaStavropoulou) [inline packages/core/tests/test_integration.py:80]: I found this way of mocking upstream assets in the docs. Not sure if there is a better way.

File: packages/floors_estimation/src/bag3d/floors_estimation/jobs.py
Comment (GinaStavropoulou) [inline packages/floors_estimation/src/bag3d/floors_estimation/jobs.py:7]: Deprecation warning.

**Comment by GinaStavropoulou** [comment]
I am going to close this because I want to add some more work with the mocks

**Comment by GinaStavropoulou** [comment]
I reopened this PR, now it is ready for review.

File: packages/common/src/bag3d/common/utils/database.py
Comment (GinaStavropoulou) [inline packages/common/src/bag3d/common/utils/database.py:10]: this is only so that this is not taken into account for the coverage estimation - I have added it in many places. It can of course be removed.

File: docs/SOURCE_DATASETS.md
Comment (GinaStavropoulou) [inline docs/SOURCE_DATASETS.md:42]: I copied all the info from the 3dbag website

File: packages/core/src/bag3d/core/assets/export/metadata.py
Comment (balazsdukai) [reply packages/core/src/bag3d/core/assets/export/metadata.py:186]: I think it is ok to keep going and write an empty, or partially empty metadata.json if one of the upstream assets were not materialized. As you say, this helps in testing.
In a prodcution setup this could happen if somone runs the the pipeline for the first time and one of the assets were never materialized. However, that would show up as a failed run in dagester for that particular asset, so that's sufficient warning imo.

**Review by balazsdukai — APPROVED** [review]
It can be merged so that the work can continue. However, there are a couple of notes.

I marked one test to fail, because it seems to me that it doesn't actually do what it meant to do.

I changed EXE_PATH_GEOF to EXE_PATH_ROOFER_RECONSTRUCT to be consistent with the names of the new geoflow-roofer.

I added the the build-tools.sh script to build all the required tools (on Ubuntu). The readme is updated to reflect the new environment variables.
Well, the script builds except one, the old geoflow that is used by tyler to do the multi-format export of the tiles. I did not include the old geolfow, because I'm not sure yet how to deal with it.
I ran the integration tests with the newly built tools on my laptop and the party walls succeed :heavy_check_mark: , the core runs until tyler would call the old geoflow for the export, so that's also :heavy_check_mark: for me, the floor estimation fails.

The floor estimation test fails with a `FAILED packages/floors_estimation/tests/test_integration.py::test_job_floors_estimation - KeyError: "None of ['identificatie'] are in the columns"` that comes from:

```
packages/floors_estimation/src/bag3d/floors_estimation/assets/floors_estimation.py:213: in preprocessed_features
    data.set_index('identificatie', inplace=True, drop=True)
```

**Pull request merged**
